### PR TITLE
rewrite themer

### PR DESCRIPTION
**Pull request by amsyarasyiq**
Conflicting with #52, and fixes of it is included as well

### CODE REVIEW

File: src/theme-init.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+const SETTINGS_DIRECTORY = externalStorageDirectory + "/AliucordRN/settings/";
+const THEME_DIRECTORY = externalStorageDirectory + "/AliucordRN/themes/";
+
+export const InvalidThemes = {
+    invalidThemes: [] as string[],
+    duplicatedThemes: [] as string[]
+};
+
+// These are handled after Aliucord loads
+export let ThemeState = {} as {
```
Comment (rushiiMachine) [inline src/theme-init.ts:29]: camelCase

File: src/theme-init.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    isApplied?: boolean;
+
+    applyFailed?: boolean;
+    reason?: ThemeErrors;
+    errorArgs?: any[];
+};
+
+export const loadedThemes: Theme[] = [];
+
+export function loadTheme(constants: ThemeConstants) {
```
Comment (rushiiMachine) [inline src/theme-init.ts:40]: confusing name, just go back to `initThemer`

File: src/theme-init.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
+import { Theme } from "./entities";
+
+export enum ThemeErrors {
+    HAS_THEME_NO_SETTINGS = "Themes exists but no settings",
```
Comment (rushiiMachine) [inline src/theme-init.ts:4]: simplify to `NO_SETTINGS`

**Review by rushiiMachine — CHANGES_REQUESTED** [review]
Also move patchTheme & theme-init.ts to src/themer/patchTheme.ts & src/themer/themerInit.ts respectively

**Comment by amsyarasyiq** [comment]
Applied suggestions

**Review by nikosszzz — APPROVED** [review]

File: src/themer/patchTheme.ts [hunk 4]
```diff
@@ -29,7 +17,10 @@ export default function patchTheme() {
                 logger.info(`Overrode theme to ${ThemeStore.theme ?? "dark"}`);
 
                 if (AMOLEDThemeManager) {
-                    if (UnsyncedUserSettingsStore.useAMOLEDTheme === 2) {
+                    if (themeState.isApplied && themeState.noAMOLED) {
+                        setAMOLEDThemeEnabledBypass(false);
+                        logger.info("Disabled AMOLED theme as it's unsupported by current custom theme.");
+                    } else if (UnsyncedUserSettingsStore.useAMOLEDTheme === 2) {
```
Comment (rushiiMachine) [inline src/themer/patchTheme.ts:23]: can't you use the AMOLED state enum

**Review by rushiiMachine — CHANGES_REQUESTED** [review]

File: src/themer/patchTheme.ts (hunk 4 above)
Comment (amsyarasyiq) [reply src/themer/patchTheme.ts]: Replaced with enum

**Review by rushiiMachine — APPROVED** [review]

**Pull request merged**